### PR TITLE
Add amp-story spec

### PR DESCRIPTION
**Pull request by newmuis**
This adds the spec for the `amp-story` extension.  The implementation is being committed in parallel.

/cc @dvoytenko

### CODE REVIEW

**Comment by choumx** [comment]
I don't think we include static assets for examples in the extension folders (so far). Some extensions have embedded examples that loads a page from a Firebase app via `amp-iframe` (e.g. [amp-accordion](https://www.ampproject.org/docs/reference/components/amp-accordion)).

/to @bpaduch For the best practice here and docs review.

**Comment by bpaduch** [comment]
Re: images in extensions folder - Haven't seen other extensions include images in the extension's folder and I would suspect you'd want to keep the extension code folder clean itself.  In the [Building an Extension doc](https://github.com/ampproject/amphtml/blob/master/contributing/building-an-amp-extension.md#element-usage-documentation-and-tests), they suggest creating an example, so I'd recommend storying those images in https://github.com/ampproject/amphtml/tree/master/examples/img there and linking to them.

File: extensions/amp-story/amp-story.md [hunk 1]
```diff
@@ -14,4 +14,373 @@ See the License for the specific language governing permissions and
...
+The following markup is a decent starting point or boilerplate. Copy this and save it to a file with a `.html` extension.
+
+```html
+<!doctype html>
+<html amp lang="en">
+  <head>
+    <meta charset="utf-8">
+    <script async src="https://stamp-prototype.appspot.com/v0.js"></script>
+    <script async custom-element="amp-story"
+        src="https://stamp-prototype.appspot.com/v0/amp-story-0.1.js"></script>
```
Comment (choumx) [inline extensions/amp-story/amp-story.md]: Update these to AMP Cache URLs.

File: extensions/amp-story/amp-story.md [hunk 2]
```diff
@@ -14,4 +14,373 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-# <a name="`amp-story`"></a> `amp-story`
\ No newline at end of file
+# <a name="`amp-story`"></a> `amp-story`
+
+**THIS IS EXPERIMENTAL AND UNDER ACTIVE DEVELOPMENT!!**
```
Comment (choumx) [inline extensions/amp-story/amp-story.md]: Most AMP extensions start with a `<table>` here containing a description, required script, status, and links to examples, etc. See others for example.

Comment (newmuis) [reply extensions/amp-story/amp-story.md]: Filed #11530

**Pull request merged**